lib/core/src/irods_service_account.cpp

```diff
@@ -0,0 +1,132 @@
+#include "irods/irods_service_account.hpp"
+
+#include "irods/irods_exception.hpp"
+#include "irods/irods_get_full_path_for_config_file.hpp"
+#include "irods/irods_kvp_string_parser.hpp"
+#include "irods/irods_log.hpp"
+#include "irods/rodsErrorTable.h"
+
+#include <cerrno>
+#include <cstdint>
+#include <cstdlib>
+#include <fstream>
+#include <memory>
+
+#include <pwd.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string.hpp>
+#include <fmt/format.h>
+
+namespace irods
+{
+    const std::string SERVICE_ACCOUNT_CONFIG_FILE("service_account.config");
+    const std::string SERTICE_ACCOUNT_KW("IRODS_SERVICE_ACCOUNT_NAME");
+
+    const size_t PWD_BUF_LEN_DEFAULT = 16384; // default size of buffer for strings in struct passwd
+    const int PWD_BUF_LEN_MINIMUM = 128; // minimum size of buffer for strings in struct passwd
+
+    // FIXME: Parsing of service_account.config isn't super robust.
+    auto is_service_account() -> bool
+    {
+        std::string service_account_username;
```

Comment:
This isn't used until much later in the function. Let's move it closer to where it is needed.